.circleci/config.yml

```
@@ -0,0 +1,46 @@
+---
+version: 2
+jobs:
+  build:
+    docker:
+      - image: library/docker
+    working_directory: /tmp/open-balena-vpn
+    environment:
+      DOCKER_IMAGE: balena/open-balena-vpn
+    branches:
+      ignore:
+        - /.+-build-[0-9a-f]+/
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Build
+          command: |
+            set -x
+            docker pull ${DOCKER_IMAGE}:${CIRCLE_BRANCH} || \
+              docker pull ${DOCKER_IMAGE}:master || true
+
+            docker build \
+              --cache-from="${DOCKER_IMAGE}:${CIRCLE_BRANCH}" \
+              --cache-from="${DOCKER_IMAGE}:master" \
+              --rm=false \
+              --pull \
+              --tag "${DOCKER_IMAGE}:${CIRCLE_SHA1}" \
+              .
+      - run:
+          name: Test
+          command: IMAGE_NAME="${DOCKER_IMAGE}:${CIRCLE_SHA1}" ./automation/test.sh
+      - deploy:
```

We need to ensure a properly tagged image is pushed, at least in the master branch (otherwise, the deploy job might come through using the previous upstream master image).
```suggestion
      - deploy:
          name: Push
          command: |
            set -x
            if [ "${CIRCLE_BRANCH}" = "master" ] && [ -n "$(git tag --list --points-at HEAD)" ]; then
              docker tag "${DOCKER_IMAGE}:${CIRCLE_SHA1}" "${DOCKER_IMAGE}:${CIRCLE_BRANCH}"
              docker push "${DOCKER_IMAGE}:${CIRCLE_BRANCH}"
            fi
      - deploy:
```
This will also require a:
```
      - run: |
          docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
```
after `- setup_remote_docker` step.

---

@mikesimos would resin CI not have pushed the docker image before the circle deploy job starts (which will've been triggered by resin CI pushing to master branch)?

---

@mikesimos also, could you add the required env vars to [the circle project](https://circleci.com/gh/balena-io/open-balena-vpn/edit#env-vars) please?

---

@wrboyce the image should be there in most cases, it should, I wouldn't really rely on that post-merge. But it might be ok post-git-tag version step.

---

I 'd include the push step here for now, (for avoiding skipping a version bump deployment). BalenaCI should be able to cover staging deployments soon, so this file will be removed.

---

> I'd include the push step here for now, (for avoiding skipping a version bump deployment). 

Sure thing, better safe than sorry!

> BalenaCI should be able to cover staging deployments soon, so this file will be removed.

I didn't realise this, awesome news!

---

@wrboyce envvars are in place :+1:

---

Thanks!